src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,227 @@
+        final AtlasMetaData atlasMetaData = new AtlasMetaData(atlasSize, true, "unknown",
+                "ProtoAtlas", "unknown", "unknown", Maps.hashMap());
+        final PackedAtlasBuilder builder = new PackedAtlasBuilder().withSizeEstimates(atlasSize)
+                .withMetaData(atlasMetaData).withName(resource.getName());
+
+        // build the atlas features
+        parsePoints(builder, protoAtlasContainer.getPointsList());
+        parseLines(builder, protoAtlasContainer.getLinesList());
+        parseAreas(builder, protoAtlasContainer.getAreasList());
+
+        return (PackedAtlas) builder.get();
+    }
+
+    /**
+     * Write an Atlas to a resource in the naive ProtoAtlas format.
+     *
+     * @param atlas
+     *            the Atlas to be written
+     * @param resource
+     *            the resource to write into
+     */
+    public void write(final Atlas atlas, final WritableResource resource)
+    {
+        final ProtoAtlasContainer.Builder protoAtlasBuilder = ProtoAtlasContainer.newBuilder();
+
+        writePointsToBuilder(atlas, protoAtlasBuilder);
+        writeLinesToBuilder(atlas, protoAtlasBuilder);
+        writeAreasToBuilder(atlas, protoAtlasBuilder);
+
+        final ProtoAtlasContainer protoAtlas = protoAtlasBuilder.build();
+        resource.writeAndClose(protoAtlas.toByteArray());
+    }
+
+    private void parseAreas(final PackedAtlasBuilder builder, final List<ProtoArea> areas)
+    {
+        final ProtoLocationConverter protoLocationConverter = new ProtoLocationConverter();
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        areas.forEach(protoArea ->
+        {
+            final long identifier = protoArea.getId();
+            final List<Location> shapePoints = protoArea.getShapePointsList().stream()
+                    .map(protoLocationConverter::convert).collect(Collectors.toList());
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = protoTagListConverter.convert(protoArea.getTagsList());
+            builder.addLine(identifier, geometry, tags);
+        });
+    }
+
+    private void parseLines(final PackedAtlasBuilder builder, final List<ProtoLine> lines)
+    {
+        final ProtoLocationConverter protoLocationConverter = new ProtoLocationConverter();
+        final ProtoTagListConverter protoTagListConverter = new ProtoTagListConverter();
+        lines.forEach(protoLine ->
+        {
+            final long identifier = protoLine.getId();
+            final List<Location> shapePoints = protoLine.getShapePointsList().stream()
+                    .map(protoLocationConverter::convert).collect(Collectors.toList());
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = protoTagListConverter.convert(protoLine.getTagsList());
+            builder.addLine(identifier, geometry, tags);
+        });
+    }
```

Comment:
`parseAreas` and `parseLines` seem to be identical apart from different parameters. Would it be worth consolidating these two and passing a generic proto object?